pkg/querier/tail.go

```diff
@@ -344,6 +344,8 @@ func newTailer(
 		tailMaxDuration:           tailMaxDuration,
 	}
 
+	t.openStreamIterator.Push(historicEntries)
```

Comment:
@sandlis In https://github.com/grafana/loki/pull/842 I'm suggesting a different path, fixing it at the root. What's your take?

Reply:
Had a quick look, looks good actually. Will check it out properly later today or tomorrow. Thanks!